### ISSUE TITLE
Translating to other languages

### ISSUE THREAD

**Issue by iafilatov** [Rlshep/BJCPBeerStyles_Android#21]
Hello Richard. Thank you for the great app!

I know someone who'd be willing to translate BJCP into Ukrainian and have it in the app as well. I've poked around the code a bit and IIUC most, if not all, strings are contained in `BjcpBeerStyles.db`.  I can provide you with the translated version of the same db. If you are willing to support the effort, could you please answer a couple of questions?
1. Is it enough to translate just the `BjcpBeerStyles.db` or are there any other strings (i.e. UI) that would need to be translated that are not in the DB?
2. I'm guessing that some kind of fall-back to English would be required for when the English DB is updated before the translated DB is. Do you think it would be possible for you to implement such fallback? If not, any hints and recommendations on how you would do it would be greatly appreciated.
3. Any other considerations re/ translation that you might have as someone who knows the code best? Anything you think worth mentioning?

**Comment by iafilatov** [comment]
I noticed that the DB strings in English and Spanish. I might be over-thinking things. If I just add Ukrainian strings with a correct language tag, will it work?

**Comment by Rlshep** [comment]
You are correct that most of the strings related to the actual styles are in the database. The best place to start on a translation is this XML file (https://github.com/Rlshep/BJCPBeerStyles_Database/blob/master/db/styleguide-2015_en.xml) file that I use to create the database. The values/strings.xml in Android project are app specific and also need translated. The other place would be Play Store description. One other consideration is any special characters and how they would be searched.

I haven't had to deal with a major version change of the guidelines yet. I'm guessing there would be considerable lag between new versions and new translations.

**Comment by Rlshep** [comment]
One other thought you and your friend may want to contact the BJCP to help do an official translation, then help update the app. Information here: http://dev.bjcp.org/international-resources/translations/

**Comment by iafilatov** [comment]
Thanks Richard, you responses are really helpful. We should be able to take it from here, at least all the XMLs are translated and ready.

**Comment by Rlshep** [comment]
Just submit a pull request when ready

**Comment by chatogithub** [comment]
Hi Richard! thank you for the reference to GitHub for sending comments about the app. I'm @Chatotuiter from Twitter :-)
I'm also BJCP Certified judge, and many of my fellow judges were surprised when finding the app completely translated to Spanish. But now it all looks "overtranslated" :-) Let me explain with an example. Between Spanish jugdes, and regular people as well, we name the beer with its original name. No need for translation there. For instance, style 22 appears as "Ale Americano Fuerte", which sounds confusing even for realizing (in a competition, for example) which style are we referring to. So, to preserve clarity, I strongly suggest to leave the style names as originally written (in this case "Strong American Ale"), and the same goes for the sub styles.
I know that this may look like "undoing the translation", but it is how we Spanish speakers recognize and name the beer styles, by their original name. Sub styles in Style 10 are a good example of how the beer names should be left untouched :-)
Sorry about the long message, and let me know if I could give you a hand for improving your great app!
Regards

**Comment by Rlshep** [comment]
Thanks for the feedback. I'm still wondering if it is just beer judges and a few beer enthusiasts that are familiar with the English names and not the general Spanish speaking population. 

Either way with the latest release in Beta I've added the option for English names. Please, sign up and provide further feedback.

**Comment by chatogithub** [comment]
Hi there!
About your thoughts on how much people are familiar with the English Names, I'd say that almost everyone... I've been in Chile, Perú, Colombia and Mexico and when you see the board of the bars with the beers offered, the styles are always in English, German, or its corresponding original name and, most important, people refers to any specific beer style (when they know it). I don't recall anyone asking or talking about a "Escocesa de exportacion" intead of a "Scottish Export", for example. :-) 
Even commercial beers show the style name as it appears originally, so again, my suggestion is to keep the original names for the main category and the subsequent subcategories. 
I'm attaching a screenshot from the BJCP 2015 Guide translated to Spanish, so you can have a closer idea of how the "English / Spanish" mixture should look like. I still didn't take a deeper view to the latest version, but if I find anything more I'll let you know, and if there is another way in which I can help you, just let me know. 
![1A](https://user-images.githubusercontent.com/50187904/57200298-9f336580-6f60-11e9-868d-4caf10c3633f.jpg)

**Comment by chatogithub** [comment]
Hi Richard, again with some comments on the app. First of all, the new setting for showing the beer styles in English is really great, thank you! 
I noticed that when using the Spanish version, with titles in English, Style 27 misses the titles for "London Bown Ale", and "Pre-Prohibition Lager". Just a minor adjust to make the app look even better! 
Regards

**Comment by Rlshep** [comment]
> Hi Richard, again with some comments on the app. First of all, the new setting for showing the beer styles in English is really great, thank you!
> I noticed that when using the Spanish version, with titles in English, Style 27 misses the titles for "London Bown Ale", and "Pre-Prohibition Lager". Just a minor adjust to make the app look even better!
> Regards

Change now in Beta version 1.2.5

**Comment by Rlshep** [comment]
@iafilatov  You might want to consider adding to the synonyms table to help with search and auto-complete. https://github.com/Rlshep/BJCPBeerStyles_Database/blob/master/db/load_synonyms.sql

**Comment by Rlshep** [comment]
@iafilatov Your change is in Alpha, send your email to BeerStyles2015@gmail.com so I can add you.
